cogs/helper.py

```diff
@@ -1,40 +1,103 @@
 import discord
 from discord.ext import commands
+from utils.config import subjects 
 
 
 class Helper(commands.Cog):
     def __init__(self, bot: commands.Bot) -> None:
-        self.bot = bot
-
+        self.bot = bot        
+        self.subjects = dict(map(int, pair.split(',')) for pair in subjects.split(';'))
+        
+        @bot.tree.context_menu(name="Toggle Pin")
+        async def pin_message(interaction: discord.Interaction, message: discord.Message):
+            # checks if the person issuing the action is a helper
+            helper_role_id = self.subjects[interaction.channel.id]
+            if not any(helper_role_id == role.id for role in interaction.user.roles):
+                return await interaction.response.send_message('Only subject helpers can pin/unpin messages.', ephemeral=True)
+            # checks if the command is being issued in a subject channel
+            if interaction.channel.id not in self.subjects.keys():
+                return await interaction.response.send_message('You may only pin messages in subject channels.', ephemeral=True)
```

Comment:
This logic seems to be repeated 3 times (pin, unpin, and context menu) - could extract it into its own helper check function